src/components/HeaderWithBackButton/index.js

```diff
@@ -0,0 +1,171 @@
+    shouldShowAvatarWithDisplay = false,
+    shouldShowBackButton = true,
+    shouldShowBorderBottom = false,
+    shouldShowCloseButton = false,
+    shouldShowDownloadButton = false,
+    shouldShowGetAssistanceButton = false,
+    shouldShowPinButton = false,
+    shouldShowStepCounter = false,
+    shouldShowThreeDotsButton = false,
+    stepCounter = null,
+    subtitle = '',
+    title = '',
+    titleColor = undefined,
+    threeDotsAnchorAlignment = {
+        horizontal: CONST.MODAL.ANCHOR_ORIGIN_HORIZONTAL.LEFT,
+        vertical: CONST.MODAL.ANCHOR_ORIGIN_VERTICAL.TOP,
+    },
+    threeDotsAnchorPosition = {
+        vertical: 0,
+        horizontal: 0,
+    },
+    threeDotsMenuItems = [],
+}) {
+    const [isDownloadButtonActive, temporarilyDisableDownloadButton] = useThrottledButtonState();
+    const {translate} = useLocalize();
+    const {isKeyboardShown} = useKeyboardState();
+    return (
+        <View style={[styles.headerBar, shouldShowBorderBottom && styles.borderBottom, shouldShowBackButton && styles.pl2]}>
+            <View style={[styles.dFlex, styles.flexRow, styles.alignItemsCenter, styles.flexGrow1, styles.justifyContentBetween, styles.overflowHidden]}>
+                {shouldShowBackButton && (
+                    <Tooltip text={translate('common.back')}>
+                        <PressableWithoutFeedback
+                            onPress={() => {
+                                if (isKeyboardShown) {
+                                    Keyboard.dismiss();
+                                }
+                                onBackButtonPress();
+                            }}
+                            style={[styles.touchableButtonImage]}
+                            accessibilityRole="button"
+                            accessibilityLabel={translate('common.back')}
+                        >
+                            <Icon
+                                src={Expensicons.BackArrow}
+                                fill={iconFill}
+                            />
+                        </PressableWithoutFeedback>
+                    </Tooltip>
+                )}
+                {shouldShowAvatarWithDisplay && (
+                    <AvatarWithDisplayName
+                        report={parentReport || report}
+                        policies={policies}
+                        personalDetails={personalDetails}
+                    />
+                )}
+                {!shouldShowAvatarWithDisplay && (
+                    <Header
+                        title={title}
+                        subtitle={stepCounter && shouldShowStepCounter ? translate('stepCounter', stepCounter) : subtitle}
+                        textStyles={titleColor ? [StyleUtils.getTextColorStyle(titleColor)] : []}
+                    />
+                )}
+                <View style={[styles.reportOptions, styles.flexRow, styles.pr5, styles.alignItemsCenter]}>
+                    {shouldShowDownloadButton && (
```

Comment:
This refactor has caused a deploy blocker https://github.com/Expensify/App/pull/23129 as we have forgot to pass the children which was a change made in the deleted file but probably did not cause morege conflicts or it was missed when resolved.

Reply:
Yes, my bad I missed that change when resolving conflicts